### PR TITLE
Fix windows build

### PR DESCRIPTION
**Pull request by ctz**
This PR adds a Cargo.lock to avoid some broken versions of windows platform crates.

Also ninja-build installation step for windows. I don't think that is strictly necessary (because nothing sets the aws-lc-rs/fips feature here), but also is very low cost.

### CODE REVIEW

**Comment by cpu** [comment]
I think the various CI `cargo` invocations should gain a `--locked` flag in this changeset (unless there was a reason you avoided that?)

**Review by djc — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -361,4 +361,4 @@ jobs:
         uses: taiki-e/install-action@cargo-hack
 
       - name: Check feature powerset
-        run: cargo hack check --feature-powerset --no-dev-deps
+        run: cargo hack check --locked --feature-powerset --no-dev-deps
```
Comment (cpu) [inline .github/workflows/ci.yml:364]: IIRC I had to explicitly leave out `--locked` for the equiv. in Rustls land:
https://github.com/rustls/rustls/blob/124f31123cfcb7c03ba29d17a1755d247b073490/.github/workflows/daily-tests.yml#L138

If I remember right it ended up stomping over its own lockfile working through the powerset.

**Review by cpu — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cpu) [reply .github/workflows/ci.yml:364]: https://github.com/rustls/rustls/commit/1a939124e8b8a72f21bdb557b8d80dc6eef72522

Comment (ctz) [reply .github/workflows/ci.yml]: Aha! Dropped this --locked too.

**Pull request merged**